ccvm_simulators/solvers/ccvm_solver.py

```diff
@@ -338,3 +348,150 @@ def machine_energy(self, machine: str, machine_parameters: dict = None):
             )
 
         return energy_method(machine_parameters)
+
+    ##############################
+    ### MACHINE TIME FUNCTIONS ###
+    ##############################
+
+    def _validate_machine_time_dataframe_columns(self, dataframe):
+        """Validates that the given dataframe contains the required columns when
+        calculating optics machine time on DL-CCVM and MF-CCVM solvers.
+
+        Args:
+            dataframe (DataFrame): The dataframe to validate.
+
+        Raises:
+            ValueError: If the given dataframe is missing any of the required columns.
+        """
+        required_columns = ["pp_time", "iterations"]
+
+        missing_columns = [
+            col for col in required_columns if col not in dataframe.columns
+        ]
```

Comment:
If the purpose of this function is to check if items in the required_columns, then wouldn't it be better to raise the ValueError as soon as the condition is not met, rather than creating a list and checking if it is empty?

Reply:
I think you are right, "ask for forgiveness" is the more pythonic strategy. This way was intended to give more feedback to the user (i.e. they can fix all the errors at once) but if we just tell them the required columns that should be enough for them to look into it themselves 👍